mempool/nop_mempool.go

```diff
@@ -0,0 +1,113 @@
+package mempool
+
+import (
+	"errors"
+
+	abcicli "github.com/cometbft/cometbft/abci/client"
+	abci "github.com/cometbft/cometbft/abci/types"
+	"github.com/cometbft/cometbft/internal/service"
+	"github.com/cometbft/cometbft/p2p"
+	"github.com/cometbft/cometbft/types"
+)
+
+// NopMempool is a mempool that does nothing.
+//
+// The ABCI app is responsible for storing, disseminating, and proposing transactions.
+// See [ADR-111](../docs/architecture/adr-111-nop-mempool.md).
+type NopMempool struct{}
```

Comment:
Is there any discrepancy between this implementation and the one described in ADR111? If yes, please adapt the ADR.

Reply:
Done in https://github.com/cometbft/cometbft/pull/1643/commits/4872946bfe6a2ea3756504df1c6924dce2fc5406